packages/nx/src/plugins/js/lock-file/lock-file.ts

```diff
@@ -187,20 +187,21 @@ function getLockFilePath(packageManager: PackageManager): string {
     return NPM_LOCK_PATH;
   }
   if (packageManager === 'bun') {
-    try {
-      const bunVersion = execSync('bun --version').toString().trim();
-      // In version 1.2.0, bun switched to a text based lockfile format by default
-      if (gte(bunVersion, '1.2.0')) {
-        return BUN_TEXT_LOCK_FILE;
-      }
-      return BUN_LOCK_PATH;
-    } catch {
-      return BUN_LOCK_PATH;
-    }
+    return isBunTextLockFile() ? BUN_TEXT_LOCK_FILE : BUN_LOCK_PATH;
   }
   throw new Error(`Unknown package manager: ${packageManager}`);
 }
 
+function isBunTextLockFile() {
+  try {
+    const bunVersion = execSync('bun --version').toString().trim();
+    // In version 1.2.0, bun switched to a text based lockfile format by default
+    return gte(bunVersion, '1.2.0');
+  } catch {
+    return false;
```

Comment:
I'm not quite sure which is the better default setting.

Reply:
In versions above 1.2.0 you can still tell bun to use old format. Probably would make more sense to just look for either file